test/common_utils.py

```diff
@@ -270,84 +270,57 @@ def combinations_grid(**kwargs):
-    def __init__(
-        self,
-        actual,
-        expected,
-        *,
-        mae=False,
-        **other_parameters,
-    ):
-        if all(isinstance(input, PIL.Image.Image) for input in [actual, expected]):
-            actual, expected = [to_image_tensor(input) for input in [actual, expected]]
+assert_equal = functools.partial(torch.testing.assert_close, rtol=0, atol=0)
 
-        super().__init__(actual, expected, **other_parameters)
-        self.mae = mae
 
-    def compare(self) -> None:
-        actual, expected = self.actual, self.expected
+def assert_close_with_image_support(actual, expected, *, mae=False, atol=None, msg=None, **kwargs):
+    def compare(actual, expected):
+        if all(isinstance(input, PIL.Image.Image) for input in [actual, expected]):
+            actual, expected = [to_image_tensor(input) for input in [actual, expected]]
 
-        self._compare_attributes(actual, expected)
-        actual, expected = self._equalize_attributes(actual, expected)
+        enable_mae_comparison = all(isinstance(input, torch.Tensor) for input in [actual, expected])
+
+        value_comparison_failure = False
+
+        def msg_callback(default_msg):
+            # This is a dirty hack that let's us "hook" into the comparison logic of `torch.testing.assert_close`. It
+            # will only be triggered in case of failed value comparison. This let's us reuse all of the attribute
+            # checking that `torch.testing.assert_close` does while giving us the ability to ignore the regular value
+            # comparison.
+            nonlocal value_comparison_failure
+            value_comparison_failure = True
+            if msg is None:
+                return default_msg
+            elif isinstance(msg, str):
+                return msg
+            elif callable(msg):
+                return msg(default_msg)
+            else:
+                raise pytest.UsageError(f"`msg` can be either be `None`, a `str` or a callable, but got {msg}.")
 
-        if self.mae:
-            if actual.dtype is torch.uint8:
-                actual, expected = actual.to(torch.int), expected.to(torch.int)
-            mae = float(torch.abs(actual - expected).float().mean())
-            if mae > self.atol:
-                self._fail(
-                    AssertionError,
-                    f"The MAE of the images is {mae}, but only {self.atol} is allowed.",
-                )
-        else:
-            super()._compare_values(actual, expected)
+        try:
+            return torch.testing.assert_close(
+                actual, expected, atol=atol, msg=msg_callback if enable_mae_comparison else msg, **kwargs
+            )
+        except AssertionError:
+            if not (value_comparison_failure and mae):
+                raise
 
+        if actual.dtype is torch.uint8:
+            actual, expected = actual.to(torch.int), expected.to(torch.int)
+        mae_value = float(torch.abs(actual - expected).float().mean())
```

Comment:
(Hopefully correct?) NIT

```suggestion
        mae_value = torch.abs(actual.float() - expected.float()).mean()
```